tests/ui/codegen/equal-pointers-unequal/as-cast/basic.rs

```diff
@@ -0,0 +1,20 @@
+//@ known-bug: #107975
+//@ run-pass
+
+fn main() {
+    let a: usize = {
+        let v = 0u8;
+        &v as *const _ as usize
+    };
+    let b: usize = {
+        let v = 0u8;
+        &v as *const _ as usize
+    };
+
+    // `a` and `b` are not equal.
+    assert_ne!(a, b);
+    // But they are the same number.
+    assert_eq!(format!("{a}"), format!("{b}"));
+    // And they are equal.
+    assert_eq!(a, b);
+}
```

Comment:
You can combine test cases with the same test conditions (such as run-pass) that do not affect each other into one `.rs` file, which can reduce the number of test files.

Reply:
As I have mentioned [above](https://github.com/rust-lang/rust/pull/127003#discussion_r1678420000), I am a bit reluctant to do so.

Reply:
That's ok